--- graph_types/graph_impl_fragment.h	2018-11-13 17:12:45.000000000 +0100
+++ graph_types/graph_impl_fragment.h	2018-11-13 17:12:06.000000000 +0100
@@ -293,7 +293,7 @@
         return true;
     if (left.len > right.len)
         return false;
-    if ((left.reversed) < (right.reversed))
+    if (left.reversed < right.reversed)
         return true;
     // if (left.reversed > right.reversed)
     //     return false;
